.circleci/config.yml

```diff
@@ -0,0 +1,596 @@
+            JAR_FILE_URL=`gsutil ls $JAR_FILE_URL_PATTERN  | sort -r | head -1`
+            if [ -z $JAR_FILE_URL ]
+              then
+                echo "Could not find archive using pattern ${JAR_FILE_URL_PATTERN}"
+                exit 1
+            fi
+            JAR_FILE_PATH="target/${JAR_BASE_NAME}.jar"
+            mkdir -p $(dirname $JAR_FILE_PATH)
+            gsutil cp  $JAR_FILE_URL $JAR_FILE_PATH
+            echo ""
+  deploy-jar:
+    parameters:
+      jar_base_name:
+        type: string
+      deploy_dir:
+        type: string
+        default: deployment-assets
+    steps:
+      - set-deployment-env
+      - render-configs
+      - run:
+          name: Gather deployment assets
+          command: |
+            DEPLOY_DIR='<<parameters.deploy_dir>>'
+            rm -rf $DEPLOY_DIR
+            mkdir -p $DEPLOY_DIR
+            cd $DEPLOY_DIR
+            cp ../target/<<parameters.jar_base_name>>.jar .
+            cp ../src/appengine/<<parameters.jar_base_name>>.yaml .
+            cp ../output-config/application.conf .
+            cp -r ../output-config/fc_keys .
+      - auth-gcp-service-account
+      - run:
+          name: Deploy <<parameters.jar_base_name>> to GAE
+          command: |
+            set -u
+            gcloud app deploy --version="${SHORT_GIT_SHA}" --stop-previous-version --project "broad-ddp-${ENVIRONMENT}" <<parameters.deploy_dir>>/<<parameters.jar_base_name>>.yaml
+      - update-deployment-sheet:
+          jar_base_name: <<parameters.jar_base_name>>
+      - run:
+          name: Delete previous versions of service
+          command: |
+            set +x
+            # extract the service name from the GAE yaml file
+            CONFIG_FILE="<<parameters.deploy_dir>>/<<parameters.jar_base_name>>.yaml"
+            SERVICE=$(grep -Eo -m1 "^service:\s*.*" "${CONFIG_FILE}" | awk '{print $2}')
+            echo "The service name is *${SERVICE}*"
+
+            # find what other versions are currently running
+            # skip first line, exclude our version, print 2nd column
+            AWK_COMMMAND="NR > 1 && !/${SHORT_GIT_SHA}/ {print \$2}"
+            VERSIONS_TO_DELETE=$(gcloud app versions list --service "${SERVICE}" --project "broad-ddp-${ENVIRONMENT}" | awk "${AWK_COMMMAND}")
+
+            # Sometime deleting versions fails. If there are less than 3 versions then we are not going to worry about it
+            # and set flag so this job step does not fail when exit error value is generated
+            if [[ ! -z $VERSIONS_TO_DELETE ]] && [[ "${#VERSIONS_TO_DELETE[@]}" -lt 3 ]]; then
+              set +e
+            fi
+
+            for CURRENT_VERSION in $VERSIONS_TO_DELETE; do
+              echo "Deleting version $CURRENT_VERSION of service $SERVICE"
+              gcloud app versions delete --quiet --service "$SERVICE" "$CURRENT_VERSION" --project "broad-ddp-${ENVIRONMENT}"
+            done;
+            :
```

Comment:
Looks like an empty line with a colon? Or are we using the bash `:` builtin?

Reply:
The builtin. Intended to be a no-op with exit code 0 . Last command in this script needs to exit with a 0. I have added a comment to clarify.
https://github.com/broadinstitute/ddp-study-server/blob/f1c2bf0c25787c26b921bed84d5a44d6ef8eae82/.circleci/config.yml#L278

Reply:
Can we just use `exit 0`?